node/network/dispute-distribution/src/sender/error.rs

```diff
@@ -20,33 +20,30 @@
 use thiserror::Error;
 
 use polkadot_node_primitives::disputes::DisputeMessageCheckError;
-use polkadot_node_subsystem_util::{runtime, Fault};
+use polkadot_node_subsystem_util::runtime;
 use polkadot_subsystem::SubsystemError;
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, derive_more::From)]
 #[error(transparent)]
-pub struct Error(pub Fault<NonFatal, Fatal>);
-
-impl From<NonFatal> for Error {
-	fn from(e: NonFatal) -> Self {
-		Self(Fault::from_non_fatal(e))
-	}
-}
-
-impl From<Fatal> for Error {
-	fn from(f: Fatal) -> Self {
-		Self(Fault::from_fatal(f))
-	}
+pub enum Error {
```

Comment:
This seems to be a pattern, so making this a generic and only having to define type aliases in the subsystems might make sense.

Reply:
It is, but due to limitations in the type system, I was not able to abstract this in a meaningful way. I haven't tried macros, there might also be ways, I was not thinking of ... so we could discuss this over some beverage at some point ;-)